checker/src/org/checkerframework/checker/index/searchindex/SearchIndexTransfer.java

```diff
@@ -0,0 +1,74 @@
+package org.checkerframework.checker.index.searchindex;
+
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.checker.index.IndexAbstractTransfer;
+import org.checkerframework.checker.index.IndexUtil;
+import org.checkerframework.checker.index.qual.SearchIndex;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+
+/**
+ * The transfer function for the SearchIndex checker. Allows SearchIndex to become NegativeIndexFor.
+ */
+public class SearchIndexTransfer extends IndexAbstractTransfer {
+
+    // The ATF (Annotated Type Factory).
+    private SearchIndexAnnotatedTypeFactory aTypeFactory;
+
+    private CFAnalysis analysis;
+
+    public SearchIndexTransfer(CFAnalysis analysis) {
+        super(analysis);
+        this.analysis = analysis;
+        aTypeFactory = (SearchIndexAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * Special binary search handling: if the left value is exactly comparison, and the right side
+     * is a search index, then convert the right side in the store to a negative index for.
+     */
```

Comment:
Please document the formal parameters, especially `comparison` which is obscure.  Should you use an enum rather than literal values?